ledger/src/blockstore.rs

```diff
@@ -1760,10 +1760,13 @@ impl Blockstore {
     }
 
     pub fn cache_block_time(&self, slot: Slot, timestamp: UnixTimestamp) -> Result<()> {
-        if !self.is_root(slot) {
-            return Err(BlockstoreError::SlotNotRooted);
+        let slot_meta_cf = self.db.column::<cf::SlotMeta>();
+        if slot_meta_cf.get(slot)?.is_none() {
```

Comment:
Can't wait for this to be released: https://github.com/rust-rocksdb/rust-rocksdb/pull/518, so that we can check for existence without eating the fetch/deserialization cost of getting the value.

Is there anything bad that can happen if the `SlotMeta` doesn't exist but the block time does? 

Just curious because it seems like this case shouldn't happen because we only add to the cached block here after freeze from replay/blockstore replay on startup via `process_single_slot()`.

Even including the snapshot root bank which we may not have the ledger block for (I don't think we call the cache block method on that bank since it's not processed via `process_single_slot()). And clean hopefully shouldn't be running on recent slots you're still replaying.

Reply:
Good point! I think you're right; there doesn't seem to be any risk of a block time existing where `SlotMeta` does not. I'll just remove this check altogether.

Reply:
Done here: https://github.com/solana-labs/solana/pull/17109/commits/0f1741c81b00f8cccc1a2077bc76915cb019cbb8

Reply:
@carllin @CriesofCarrots 

> Can't wait for this to be released: rust-rocksdb/rust-rocksdb#518

This is finally landed in upstream and available in our codebase starting from: https://github.com/solana-labs/solana/pull/18904

Reply:
... but it look like the new method incurs the possibility of false positive:

https://docs.rs/rocksdb/0.17.0/rocksdb/struct.DBWithThreadMode.html#method.key_may_exist_cf_opt:

> Returns false if the given key definitely doesn’t exist in the specified column family, otherwise returns true.